```bash

Seems to have failed CI, gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make[2]: Entering directory `/home/iojs/build/workspace/node-test-commit-linux-abi/nodes/ubuntu1404-64/test/addons/02_object_factory/build'
  CXX(target) Release/obj.target/addon/addon.o
../binding.cc: In function 'void CreateFunction(napi_env, napi_callback_info)':
../binding.cc:26:48: error: 'napi_set_function_name' was not declared in this scope
   status = napi_set_function_name(env, fn, name);
                                                ^
../binding.cc: In function 'void Init(napi_env, napi_value, napi_value)':
../binding.cc:35:63: warning: missing initializer for member 'napi_property_descriptor::getter' [-Wmissing-field-initializers]
   napi_property_descriptor desc = { "exports", CreateFunction };
                                                               ^
../binding.cc:35:63: warning: missing initializer for member 'napi_property_descriptor::setter' [-Wmissing-field-initializers]
../binding.cc:35:63: warning: missing initializer for member 'napi_property_descriptor::value' [-Wmissing-field-initializers]
../binding.cc:35:63: warning: missing initializer for member 'napi_property_descriptor::attributes' [-Wmissing-field-initializers]
../binding.cc:35:63: warning: missing initializer for member 'napi_property_descriptor::data' [-Wmissing-field-initializers]
make[2]: *** [Release/obj.target/binding/binding.o] Error 1
make[2]: Leaving directory `/home/iojs/build/workspace/node-test-commit-linux-abi/nodes/ubuntu1404-64/test/addons-abi/5_function_factory/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/iojs/build/workspace/node-test-commit-linux-abi/nodes/ubuntu1404-64/deps/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:192:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Linux 3.13.0-71-generic
gyp ERR! command "/home/iojs/build/workspace/node-test-commit-linux-abi/nodes/ubuntu1404-64/out/Release/node" "/home/iojs/build/workspace/node-test-commit-linux-abi/nodes/ubuntu1404-64/deps/npm/node_modules/node-gyp/bin/node-gyp" "rebuild" "--python=/usr/bin/python" "--directory=/home/iojs/build/workspace/node-test-commit-linux-abi/nodes/ubuntu1404-64/test/addons-abi/5_function_factory/" "--nodedir=/home/iojs/build/workspace/node-test-commit-linux-abi/nodes/ubuntu1404-64"
gyp ERR! cwd /home/iojs/build/workspace/node-test-commit-linux-abi/nodes/ubuntu1404-64/test/addons-abi/5_function_factory
gyp ERR! node -v v8.0.0-pre
gyp ERR! node-gyp -v v3.5.0
gyp ERR! not ok 
make[1]: *** [test/addons-abi/.buildstamp] Error 1
make[1]: *** Waiting for unfinished jobs....
```

--- a/src/node_jsvmapi.cc
+++ b/src/node_jsvmapi.cc
@@ -644,9 +644,9 @@ napi_status napi_get_current_env(napi_env* e) {
 
 napi_status napi_create_function(
     napi_env e,
+    const char* utf8name,
     napi_callback cb,
     void* data,
-    const char* utf8name,
     napi_value* result) {
   NAPI_PREAMBLE(e);
   CHECK_ARG(result);

--- a/src/node_jsvmapi.cc
+++ b/src/node_jsvmapi.cc
@@ -646,7 +646,7 @@ napi_status napi_create_function(
     napi_env e,
     napi_callback cb,
     void* data,
-    napi_propertyname name,
+    const char* utf8name,
     napi_value* result) {
   NAPI_PREAMBLE(e);
   CHECK_ARG(result);
@@ -666,10 +666,11 @@ napi_status napi_create_function(
     isolate, v8impl::FunctionCallbackWrapper::Invoke, cbdata);
 
   retval = scope.Escape(tpl->GetFunction());
-
-  if (name) {
-    v8::Local<v8::String> n = v8impl::V8LocalStringFromJsPropertyName(name);
-    retval->SetName(n);
+  
+  if (utf8name) {
+    v8::Local<v8::String> namestring;
+    CHECK_NEW_FROM_UTF8(isolate, namestring, utf8name);
+    retval->SetName(namestring);
   }
 
   *result = v8impl::JsValueFromV8LocalValue(retval);

--- a/src/node_jsvmapi.h
+++ b/src/node_jsvmapi.h
@@ -108,9 +108,9 @@ NODE_EXTERN napi_status napi_create_symbol(napi_env e,
                                            const char* s,
                                            napi_value* result);
 NODE_EXTERN napi_status napi_create_function(napi_env e,
+                                             const char* utf8name,
                                              napi_callback cb,
                                              void* data,
-                                             const char* utf8name,
                                              napi_value* result);
 NODE_EXTERN napi_status napi_create_error(napi_env e,
                                           napi_value msg,

--- a/src/node_jsvmapi.h
+++ b/src/node_jsvmapi.h
@@ -110,7 +110,7 @@ NODE_EXTERN napi_status napi_create_symbol(napi_env e,
 NODE_EXTERN napi_status napi_create_function(napi_env e,
                                              napi_callback cb,
                                              void* data,
-                                             napi_propertyname name,
+                                             const char* utf8name,
                                              napi_value* result);
 NODE_EXTERN napi_status napi_create_error(napi_env e,
                                           napi_value msg,

--- a/test/addons-abi/5_function_factory/binding.cc
+++ b/test/addons-abi/5_function_factory/binding.cc
@@ -15,7 +15,7 @@ void CreateFunction(napi_env env, napi_callback_info info) {
   napi_status status;
 
   napi_value fn;
-  status = napi_create_function(env, MyFunction, nullptr, "theFunction", &fn);
+  status = napi_create_function(env, "theFunction", MyFunction, nullptr, &fn);
   if (status != napi_ok) return;
 
   status = napi_set_return_value(env, info, fn);

--- a/test/addons-abi/5_function_factory/binding.cc
+++ b/test/addons-abi/5_function_factory/binding.cc
@@ -14,16 +14,12 @@ void MyFunction(napi_env env, napi_callback_info info) {
 void CreateFunction(napi_env env, napi_callback_info info) {
   napi_status status;
 
-  napi_value fn;
-  status = napi_create_function(env, MyFunction, nullptr, nullptr, &fn);
-  if (status != napi_ok) return;
-
   napi_propertyname name;
   status = napi_property_name(env, "theFunction", &name);
   if (status != napi_ok) return;
 
-  // omit this to make it anonymous
-  status = napi_set_function_name(env, fn, name);
+  napi_value fn;
+  status = napi_create_function(env, MyFunction, nullptr, name, &fn);
   if (status != napi_ok) return;
 
   status = napi_set_return_value(env, info, fn);
@@ -32,7 +28,7 @@ void CreateFunction(napi_env env, napi_callback_info info) {
 
 void Init(napi_env env, napi_value exports, napi_value module) {
   napi_status status;
-  napi_property_descriptor desc = { "exports", CreateFunction };
+  napi_property_descriptor desc = { "exports", CreateFunction, nullptr, nullptr, nullptr, napi_default, nullptr };
   status = napi_define_properties(env, module, 1, &desc);
   if (status != napi_ok) return;
 }

--- a/test/addons-abi/5_function_factory/binding.cc
+++ b/test/addons-abi/5_function_factory/binding.cc
@@ -1,4 +1,4 @@
-#include <node_jsvmapi.h>
+﻿#include <node_jsvmapi.h>
 
 void MyFunction(napi_env env, napi_callback_info info) {
   napi_status status;
@@ -14,12 +14,8 @@ void MyFunction(napi_env env, napi_callback_info info) {
 void CreateFunction(napi_env env, napi_callback_info info) {
   napi_status status;
 
-  napi_propertyname name;
-  status = napi_property_name(env, "theFunction", &name);
-  if (status != napi_ok) return;
-
   napi_value fn;
-  status = napi_create_function(env, MyFunction, nullptr, name, &fn);
+  status = napi_create_function(env, MyFunction, nullptr, "theFunction", &fn);
   if (status != napi_ok) return;
 
   status = napi_set_return_value(env, info, fn);

